aten/src/ATen/native/RNN.h

```diff
@@ -19,5 +19,30 @@ DECLARE_DISPATCH(rnn_packed_fn, gru_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_tanh_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_relu_packed_cudnn_stub);
 
-}} // namespace at::native
 
+void check_device(const Tensor& input, const TensorList& hx, const TensorList& params) {
+  auto input_device = input.device();
+  bool input_device_is_cuda = input_device.is_cuda();
+  for (auto h : hx) {
+    auto h_device = h.device();
+    bool h_device_is_cuda = h_device.is_cuda();
+    AT_CHECK((input_device_is_cuda && h_device_is_cuda) || (!input_device_is_cuda && !h_device_is_cuda), "Input and Hidden tensors must share the same device.");
```

Comment:
Could these error messages be more explicit and state what devices they are currently on?

Reply:
Just saying something like we expect them to be on the same device, but found A is on xxx and B is on yyy should be fine.